core/locale/infrastructure/translation_service.go

```diff
@@ -123,33 +134,43 @@ func (ts *TranslationService) translateWithLib(localeCode string, key string, co
 	}
 	return label, nil
 }
-func (ts *TranslationService) loadFiles() {
-	if ts.filesLoaded && !ts.devmode {
+
+func (ts *TranslationService) reloadFilesIfNecessary() {
+	if !ts.devmode {
 		return
 	}
 
-	if ts.translationFile != "" {
-		err := ts.i18bundle.LoadTranslationFile(ts.translationFile)
+	var lastFileChange time.Time
+	for _, fileName := range ts.translationFiles {
+		stat, err := os.Stat(fileName)
 		if err != nil {
-			ts.logger.Warn(fmt.Sprintf("Load translationfile failed: %s", err))
+			continue
 		}
-	}
-	if len(ts.translationFiles) > 0 {
-		for _, file := range ts.translationFiles {
-			if fileName, ok := file.(string); ok {
-				err := ts.i18bundle.LoadTranslationFile(fileName)
-				if err != nil {
-					ts.logger.Warn(fmt.Sprintf("Load translationfile failed: %s", err))
-				}
-			}
+
+		if stat.ModTime().After(lastFileChange) {
+			lastFileChange = stat.ModTime()
 		}
 	}
-	ts.filesLoaded = true
+
+	ts.RLock()
+	lastReload := ts.lastReload
+	ts.RUnlock()
+
+	if lastFileChange.After(lastReload) {
+		ts.loadFiles()
+	}
```

Comment:
This part is still not perfect. Actually we should reload the files without giving away the lock. But as far as I know, Go has no way to upgrade the read lock to a write lock. If you have any ideas how to improve this, please let me know. 
It's not too bad though. Worst case is that we reload the translations again without file changes. As it only happens in debug mode, I think it's okay for now.

Reply:
You could prevent this by using github.com/golang/groupcache/singleflight as we do it in core/cache: While loading a cache entry from the backend, all other incoming load requests wait for the result instead of load it again.

But I'm not sure if this case is worth the effort.

Reply:
I played around with the singleflight for a while, but couldn't get the synchronization right. In the end, I decided to move away from the R/W lock and just use a normal mutex. Even without parallel reads it's still much faster than before.

Reply:
nice, I totally agree